tf2_ros/include/tf2_ros/message_filter.h

```diff
@@ -668,7 +670,8 @@ class MessageFilter : public MessageFilterBase, public message_filters::SimpleFi
   V_string target_frames_; ///< The frames we need to be able to transform to before a message is ready
   std::string target_frames_string_;
   boost::mutex target_frames_mutex_; ///< A mutex to protect access to the target_frames_ list and target_frames_string.
-  uint32_t queue_size_; ///< The maximum number of messages we queue up
+  boost::shared_mutex cbqueue_mutex_; ///< A mutex protecting calls from callback queues
```

Comment:
It looks like this changes the size of `CBQueueCallback`. I'm not totally sure if the ABI of `CBQueueCallback` matters, but I don't have time to look deeper at this. I suspect this PR is unlikely to make it in before Noetic eol.

That's probably pretty disappointing given #538 being closed for Melodic going EOL, but not much we can do about it now.

Reply:
Thanks for flagging that @sloretz I think it's ok. It looks like a private struct inside the templated class. It's only directly used to push the data which is dynamically allocated into the queue on either `messageReady` or `messageDropped` and then the rest of the class only accesses the private queue of the `MessageFIlter` class member pointer `callback_queue_` which is then only leverages the generic interface. 

Because it's all in private elements and compiled into the implementation by the template I believe that this should not break ABI as there's no external interface accessible to this datatype, and any instantiation of this template should not be interacting with other instances in other compile units.

Reply:
Thanks for reviewing Tully! I double checked, and I think you're correct. Merging.